<?xml version="1.0"?>
<launch>
  <param name="use_sim_time" value="false"/>

  <node pkg="jetracer" type="jetracer" name="jetracer"  output="screen" respawn="true">
    <param name="port_name" value="/dev/ttyACM0"/>
    <param name="linear_correction" value="1.0"/>
    <param name="coefficient_a" value="-0.016073"/>
    <param name="coefficient_b" value="0.176183"/>
    <param name="coefficient_c" value="-23.428084"/>
    <param name="coefficient_d" value="1500"/>
    <remap from="/odom" to="/odom_raw"/>
    <param name="publish_odom_transform" value="false"/>
  </node>

	<node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf" output="screen">
    <param name="output_frame" value="odom"/>
		<param name="base_footprint_frame" value="base_footprint"/>
    <param name="freq" value="30.0"/>
    <param name="sensor_timeout" value="0.5"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="false"/>
    <param name="debug" value="false"/>
    <param name="self_diagnose" value="false"/>
    <remap from="odom" to="/odom_raw"/>
    <remap from="/imu_data" to="/imu"/>
		<remap from="/robot_pose_ekf/odom_combined" to="/odom_combined"/>
  </node>

  <node pkg="jetracer" type="odom_ekf.py" name="odom_ekf_node" output="screen"/>
  
	<node pkg="tf" type="static_transform_publisher" name="base_footprint_to_imu" args="0 0 0.02 0 0 0 base_footprint base_imu_link 20"/>

</launch>